jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -587,10 +586,7 @@ ecma_builtin_array_prototype_object_push (ecma_value_t this_arg, /**< this argum
                     ecma_builtin_array_prototype_helper_set_length (obj_p, n),
                     ret_value);
 
-    ecma_number_t *ret_num_p = ecma_alloc_number ();
-    *ret_num_p = n;
-
-    ret_value = ecma_make_number_value (ret_num_p);
+    ret_value = ecma_make_number_value (n);
```

Comment:
That is true for Array objects, but not for other Objects. They don't have a limit on their length property, and when push() is called on a non-Array object, it may go above 2^32. There is a test for this in test262.